### PR TITLE
Fare-Calculation-updated

### PR DESCRIPTION
**Pull request by gandhinikhil**
New Features has updated for Fare calculation.

### CODE REVIEW

File: src/main/java/com/example/cabbooking/constant/FareConstant.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.example.cabbooking.constant;
+
+public class FareConstant {
+
+    public static final double BASE_FARE = 50;
+    public static final double PER_KM_RATE= 10;
+    public  static final double TAX_RATE = 0.05;
+
```
Comment (sanskrati3) [inline src/main/java/com/example/cabbooking/constant/FareConstant.java]: remove extra space

File: src/main/java/com/example/cabbooking/controller/CabBookingController.java [hunk 2]
```diff
@@ -47,12 +48,12 @@ public ResponseEntity<String> addDriver(@RequestBody DriverDetailsVO driverDetai
     }
 
     @GetMapping("/find_ride")
```
Comment (sanskrati3) [inline src/main/java/com/example/cabbooking/controller/CabBookingController.java]: please read this and use proper naming convension for API endpoints 
https://pratapsharma.io/api-naming-convention

File: src/main/java/com/example/cabbooking/controller/CabBookingController.java [hunk 3]
```diff
@@ -47,12 +48,12 @@ public ResponseEntity<String> addDriver(@RequestBody DriverDetailsVO driverDetai
...
-    public ResponseEntity<List<Driver>> findRide(@RequestParam String username, @RequestParam List<Integer> source, @RequestParam List<Integer> destination) {
+    public ResponseEntity<List<RideDetailsVO>> findRide(@RequestParam String username, @RequestParam List<Integer> source, @RequestParam List<Integer> destination) {
          try{
            
-                List<Driver>availableDrivers =rideService.findRide(username, new LocationDetailsVO(source.get(0), source.get(1)), new LocationDetailsVO(destination.get(0), destination.get(1)));
+                List<RideDetailsVO>availableDrivers =rideService.findRide(username, new LocationDetailsVO(source.get(0), source.get(1)), new LocationDetailsVO(destination.get(0), destination.get(1)));
                 if(availableDrivers.isEmpty()) {
-                    ResponseEntity<List<Driver>> listResponseEntity = new ResponseEntity<>(HttpStatus.NO_CONTENT);
+                    ResponseEntity<List<RideDetailsVO>> listResponseEntity = new ResponseEntity<>(HttpStatus.NO_CONTENT);
```
Comment (sanskrati3) [inline src/main/java/com/example/cabbooking/controller/CabBookingController.java]: please add null checks

File: src/main/java/com/example/cabbooking/servicetest/DriverService.java [hunk 4]
```diff
@@ -27,11 +33,13 @@ public void addDriver(DriverDetailsVO driverDetailsVO) {
         System.out.println(driver.getName()+" "+driver.getGender()+" "+driver.getAge()+" "+driver.isAvailable()+" "+driver.getCurrentLocation()+" "+driver.getVehicle());
     }
 
-    public List<Driver> findAvailableDrivers(LocationDetailsVO source, int maxDistance) {
-        List<Driver> availableDrivers = new ArrayList<>();
+    public List<RideDetailsVO> findAvailableDrivers(LocationDetailsVO source,LocationDetailsVO destination, int maxDistance) {
+        List<RideDetailsVO> availableDrivers = new ArrayList<>();
         for (Driver driver : drivers.values()) {
             if (driver.isAvailable() && driver.getCurrentLocation().distanceTo(source) <= maxDistance) {
```
Comment (sanskrati3) [inline src/main/java/com/example/cabbooking/servicetest/DriverService.java:39]: add logs for easy debugging

**Review by sanskrati3 — CHANGES_REQUESTED** [review]
please make changes before merging

File: src/main/java/com/example/cabbooking/controller/CabBookingController.java (hunk 2 above)
Comment (gandhinikhil) [reply src/main/java/com/example/cabbooking/controller/CabBookingController.java]: Will taken care and now ressolved.

File: src/main/java/com/example/cabbooking/constant/FareConstant.java (hunk 1 above)
Comment (gandhinikhil) [reply src/main/java/com/example/cabbooking/constant/FareConstant.java]: Remove extra spaces

File: src/main/java/com/example/cabbooking/controller/CabBookingController.java (hunk 3 above)
Comment (gandhinikhil) [reply src/main/java/com/example/cabbooking/controller/CabBookingController.java]: Add null checks as required

File: src/main/java/com/example/cabbooking/servicetest/DriverService.java (hunk 4 above)
Comment (gandhinikhil) [reply src/main/java/com/example/cabbooking/servicetest/DriverService.java:39]: Logs are added

**Comment by gandhinikhil** [comment]
New Features has updated. Fare Calculation.

**Pull request merged**